core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -290,8 +292,10 @@ private int readVersionHint() {
         return Integer.parseInt(in.readLine().replace("\n", ""));
       }
 
-    } catch (IOException e) {
-      throw new RuntimeIOException(e, "Failed to get file system for path: %s", versionHintFile);
```

Comment:
Should we move `Util.getFs` out of this try/catch block? An `IOException` from not being able to load the FS should probably fail instead of returning a hint.

Reply:
Done